src/Uart/Uart.h

```diff
@@ -0,0 +1,311 @@
+    /// \brief Target baud rate
+    Uart_BaudRate baudRate;
+    /// \brief Indicator of the baud rate clock source
+    uint32_t baudRateClkSrc;
+    /// \brief Baud rate clock source frequency
+    uint32_t baudRateClkFreq;
+} Uart_Config;
+
+/// \brief A function serving as a callback called at the end of transmission.
+typedef void (*UartTxEndCallback)(void* arg);
+
+/// \brief A descriptor of an end-of-transmission event handler.
+typedef struct
+{
+    UartTxEndCallback callback; ///< Callback function
+    void* arg;                  ///< Argument to the callback function
+} Uart_TxHandler;
+
+/// \brief A function serving as a callback called upon a reception of a byte
+///        if the reception queue contains at least a number of bytes specified
+///        in the handler descriptor.
+typedef void (*UartRxEndLengthCallback)(void* arg);
+
+/// \brief A function serving as a callback called upon a reception of a byte if
+///        byte matches a target specified in the handler descriptor.
+typedef void (*UartRxEndCharacterCallback)(void* arg);
+
+/// \brief A descriptor of a byte reception event handler.
+typedef struct
+{
+    /// \brief Callback called when reception queue data count is greater
+    /// than or equal targetLength
+    UartRxEndLengthCallback lengthCallback;
+    /// \brief Callback called when a targetCharacter is received
+    UartRxEndCharacterCallback characterCallback;
+    /// \brief Argument for the length callback
+    void* lengthArg;
+    /// \brief Argument for the character callback
+    void* characterArg;
+    /// \brief Target character, upon reception of which character callback
+    /// is called
+    uint8_t targetCharacter;
+    /// \brief Target length of reception queue, upon reaching of which
+    /// length callback is called
+    uint32_t targetLength;
+} Uart_RxHandler;
+
+/// \brief A function serving as a callback called upon detection of an error by
+///        hardware.
+typedef void (*UartErrorCallback)(void* arg);
+
+/// \brief A descriptor of an error handler.
+typedef struct
+{
+    UartErrorCallback callback; ///< Callback function
+    void* arg;                  ///< Argument to the callback function
+} Uart_ErrorHandler;
+
+/// \brief Uart error flags.
+typedef struct
+{
+    bool hasOverrunOccurred;         //< Hardware FIFO overrun detected
+    bool hasFramingErrorOccurred;    //< Framing error detected
+    bool hasParityErrorOccurred;     //< Parity error detected
+    bool hasRxFifoFullErrorOccurred; //< Rx FIFO full error detected
+} Uart_ErrorFlags;
+
+/// \brief Uart device descriptor.
+typedef struct
+{
+    Uart_Id id;                     ///< Device identifier
+    Uart_TxHandler txHandler;       ///< End-of-transmission handler descriptor
+    Uart_RxHandler rxHandler;       ///< Reception handler descriptor
+    Uart_ErrorHandler errorHandler; ///< Error handler descriptor
+    ByteFifo* txFifo;               ///< Pointer to a transmission byte queue
```

Comment:
These Fifos should be initialized to NULL (understood as empty, no async transmission)

Reply:
done